packages/cadl-python/src/emitter.ts

```diff
@@ -1007,23 +1006,25 @@ function emitUnion(context: DpgContext, type: Union): Record<string, any> {
     const nonNullOptions = [...type.variants.values()].map((x) => x.type).filter((t) => !isNullType(t));
 
     const notLiteral = (t: Type): boolean => ["Boolean", "Number", "String"].indexOf(t.kind) < 0;
-    if (nonNullOptions.length > 1) {
-        if (nonNullOptions.every(notLiteral)) {
-            // Generate as CombinedType if non of the options is Literal.
-            const unionName = `MyCombinedType`;
-            return {
-                name: unionName,
-                snakeCaseName: camelToSnakeCase(unionName),
-                description: `Type of ${unionName}`,
-                isPublic: false,
-                type: "combined",
-                types: nonNullOptions.map((x) => emitType(context, x)),
-                xmlMetadata: {},
-            };
-        } else if (nonNullOptions.some(notLiteral)) {
-            // Can't generate if this union is a mixed up of literals and sub-types
-            throw Error(`Can't do union for ${JSON.stringify(nonNullOptions)}`);
+    if (nonNullOptions.every(notLiteral)) {
+        if (nonNullOptions.length === 1) {
+            // Generate as internal type if there is only one internal type in this Union.
```

Comment:
what is an internal type? not sure what we would actually generate in this case can you give an example? thanks!

Reply:
Yes @iscai-msft , for below case:
```
prop: string|null;
```
The `nonNullOptions` here will contain only one type which is `string`.

In this case, the internal type is `string`, and we just generate as that type as discussed in https://github.com/Azure/autorest.python/issues/1751